common/internal_model/src/main/scala/weco/catalogue/internal_model/index/WorksAnalysis.scala

```diff
@@ -8,12 +8,11 @@ object WorksAnalysis {
   val slashesCharFilter =
     MappingCharFilter("slashes_char_filter", mappings = Map("/" -> " __"))
 
-  // This analyzer "keeps" the hyphen, by removing it and treating hyphenated
-  // tokens as a single token.
-  val hyphensCharFilter =
-    PatternReplaceCharFilter(
-      "hyphens_char_filter",
-      pattern = "-",
+  // This char filter removes all punctuation
+  val punctuationTokenFilter =
+    PatternReplaceTokenFilter(
+      "punctuation",
+      pattern = "[^0-9\\p{L}\\s]",
```

Comment:
Maybe add a comment about why we're using `\p{L}` instead of `\w`? I think you can point back to this discussion, let's just make sure we capture it.